File: C++/Problems/Problems/problem_1798.cpp
#include "problem_1798.h"

bool problem_1798::test() {
    std::vector coins  { 1, 3 };

    int expected = 2;

    auto result = getMaximumConsecutive(coins);

    std::cout << result << std::endl;

    return result == expected;
}

int problem_1798::getMaximumConsecutive(std::vector<int> &coins) {
    std::sort(coins.begin(), coins.end());
    int result = 1;
    for (auto c : coins)
    {
        if (result < c)
            break;
        result += c;
    }

    return result;
}

File: Python/Problems/Problems/problem1798.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter

from .problem import Problem

class Problem1798(Problem):
    def test(self) -> bool:
        coins = [1, 3]

        expected = 2

        result = self.getMaximumConsecutive(coins)

        print(result)

        return result == expected

    def getMaximumConsecutive(self, coins: List[int]) -> int:
        coins.sort()
        result = 1
        for c in coins:
            if result < c:
                break
            result += c

        return result